data_service/adapters/storage/local.py

```diff
@@ -11,53 +12,87 @@
-            )
-            full_path = (
-                full_path if os.path.isdir(full_path)
-                else f"{full_path}.parquet"
-            )
-        else:
-            data_versions = self.__get_data_versions(version)
-            if data_structure_name not in data_versions:
-                raise NotFoundException(
-                    "No such data structure in data_versions file "
-                    f"for version {version}"
+        if version.startswith("0.0.0"):
+            full_path = self._get_draft_file_path(path_prefix, dataset_name)
+            if full_path is None:
+                latest = self._get_latest_version()
+                file_name = self._get_file_name_from_data_versions(
+                    latest, dataset_name
                 )
-            parquet_file_path = data_versions[data_structure_name]
-
-            full_path = (
-                f"{path_prefix}/{parquet_file_path}"
+                full_path = f'{path_prefix}/{file_name}'
+        else:
+            file_name = self._get_file_name_from_data_versions(
+                version, dataset_name
             )
+            full_path = f'{path_prefix}/{file_name}'
 
         if not os.path.exists(full_path):
-            self.log.error(f"Path {full_path} does not exist")
-            raise NotFoundException("No such data structure")
+            self.log.error(f'Path {full_path} does not exist')
+            raise NotFoundException(
+                f'No file exists for {dataset_name} in {version}'
+            )
 
         return full_path
 
-    def __get_data_versions(self, version: str) -> str:
-        version = self.__to_underscored_version(version)
+    def _get_file_name_from_data_versions(
+        self, version: str, dataset_name: str
+    ) -> str:
+        file_version = self._to_underscored_two_number_version(version)
         data_versions_file = (
             f"{self.settings.DATASTORE_DIR}/datastore"
-            f"/data_versions__{version}.json"
+            f"/data_versions__{file_version}.json"
         )
         with open(data_versions_file, encoding="utf-8") as f:
-            return json.load(f)
+            data_versions = json.load(f)
+
+        if dataset_name not in data_versions:
+            raise NotFoundException(
+                f"No {dataset_name} in data_versions file "
+                f"for version {version}"
+            )
+        return data_versions[dataset_name]
+
+    def _get_draft_file_path(
+        self, path_prefix: str, dataset_name: str
+    ) -> Union[None, str]:
+        parquet_path = f"{path_prefix}/{dataset_name}__DRAFT.parquet"
+        partitioned_parquet_path = parquet_path.replace('.parquet', '')
+        if os.path.isfile(parquet_path):
+            return parquet_path
+        elif os.path.isdir(partitioned_parquet_path):
+            return partitioned_parquet_path
+        else:
+            return None
+
+    def _get_latest_version(self):
+        datastore_files = os.listdir(
```

Comment:
👍

Reply:
@pawbu I am sure we need to do a minor change to migration scripts, I note this, thanks!